storage.go

```diff
@@ -484,6 +493,42 @@ func (s *Storage) write(ctx context.Context, path string, r io.Reader, size int6
 	return size, nil
 }
 
+func (s *Storage) writeAppend(ctx context.Context, o *Object, r io.Reader, size int64, opt pairStorageWriteAppend) (n int64, err error) {
+	rp := o.GetID()
+
+	offset, ok := o.GetAppendOffset()
+	if !ok {
+		err = fmt.Errorf("append offset is not set")
+		return
+	}
+
+	input := &service.AppendObjectInput{
+		Position:      &offset,
+		ContentLength: &size,
+		Body:          io.LimitReader(r, size),
+	}
+	if opt.HasContentMd5 {
+		input.ContentMD5 = &opt.ContentMd5
+	}
+	if opt.HasContentType {
+		input.ContentType = &opt.ContentType
+	}
+	if opt.HasStorageClass {
+		input.XQSStorageClass = &opt.StorageClass
+	}
+
+	output, err := s.bucket.AppendObjectWithContext(ctx, rp, input)
+	if err != nil {
+		return
+	}
+
+	offset = *output.XQSNextAppendPosition
```

Comment:
Maybe we need to check it before use. The value could be a `nil`.

Reply:
I wonder whether the current data uploaded successfully when the err is nil but the XQSNextAppendPosition is also nil. And what kind of return values are more appropriate in this situation?

Reply:
Yes, it's won't happen in the normal situation because we know the behavior of `qingstor` service and it's SDK. But we can't trust them because the behavior could be changed in the future (like API changed or a bug in SDK). 

So check the value is a safeguard: No matter how the behavior changes, it's won't lead `panic` in our logic and our users applications.

I plan to formalize the errors returned in service, for now, let's just returns a `errors.New("next append position is empty")`.